### PR TITLE
Add support for Cosmos Tokenizers

### PR DESCRIPTION
**Pull request by jojennin**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: This creates a new collection under common/video_tokenizers. Specifically, this collection creates a NeMo PyTorch lightning model for the Cosmos Tokenizers.

# Changelog 
- Add a PyTorch lightning model for the Cosmos Tokenizers
- Add an example for multi-modal autoregressive image generation with Cosmos Tokenizer
- Add an example for preparing a megatron-energon dataset with Cosmos Tokenizer

# Usage
* We provide README's under the following directories
    * `nemo/collections/common/video_tokenizers`
    * `nemo/collections/diffusion/data`
    * `nemo/collections/multimodal_autoregressive/data`

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/common/video_tokenizers/utils.py [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Utility functions for the inference libraries."""
+
+import os
+from glob import glob
+from typing import Any
```
[the remark this answers is not in the document]
Comment (jojennin) [reply nemo/collections/common/video_tokenizers/utils.py]: Removed in the [latest commit](https://github.com/NVIDIA/NeMo/pull/11177/commits/6b6aba0b7bc07ac98f92628ad329e8434952d04a)

File: nemo/collections/common/video_tokenizers/utils.py [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+
+import mediapy as media
+import numpy as np
+import torch
+from PIL import Image
```
[the remark this answers is not in the document]
Comment (jojennin) [reply nemo/collections/common/video_tokenizers/utils.py]: Removed in the [latest commit ](https://github.com/NVIDIA/NeMo/pull/11177/commits/6b6aba0b7bc07ac98f92628ad329e8434952d04a)

File: nemo/collections/diffusion/data/prepare_energon_dataset.py [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import pickle
+from typing import Callable, List
```
[the remark this answers is not in the document]
Comment (jojennin) [reply nemo/collections/diffusion/data/prepare_energon_dataset.py]: Resolved in commit [5b634d0dccd742be280b498831b5467bd62bb831](5b634d0dccd742be280b498831b5467bd62bb831)

File: nemo/collections/multimodal_autoregressive/data/preprocess_coyo_emu3_tokenizer.py [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+        help='The maximum number of pixels in the image. Picture will be downsampled if smaller and resize_image is set to False',
+    )
+    parser.add_argument(
+        '--chunk_stride_size', type=int, default=64, help='the stride size for neighbor chunks used for retrieval'
+    )
+
+    args = parser.parse_args()
+
+    rank = int(os.environ['LOCAL_RANK'])
+    world_size = torch.cuda.device_count()
```
[the remark this answers is not in the document]
Comment (jojennin) [reply nemo/collections/multimodal_autoregressive/data/preprocess_coyo_emu3_tokenizer.py]: Resolved in commit [5b634d0dccd742be280b498831b5467bd62bb831](5b634d0dccd742be280b498831b5467bd62bb831)

**Review by ethanhe42 — APPROVED** [review]
lgtm

**Review by zpx01 — APPROVED** [review]
lgtm
